### ISSUE TITLE
Major Mode Segment

### ISSUE THREAD

**Issue by domtronn** [domtronn/spaceline-all-the-icons.el#15]
_(Quoted from @jwintz #7)_
>And as a last point, M-x space line-toggle-major-mode-on does have any effect whatsoever. Miss it.

**Comment by domtronn** [comment]
So part of the original inspiration for this package was [`mode-icons`](https://github.com/ryuslash/mode-icons) but obviously using icon fonts rather than XPM images. And as such, the `major-mode` segment is now replaced by `spaceline-all-the-icons-mode-icon` Which _should_ replace the major mode string with the corresponding icon... However, if it can't find the icon, the major mode string appears in `spaceline-all-the-icons-process`

There's probably a little bit of work to make sure this isn't duplicated, but I'm happy to add this segment back in if you find it useful 😃

**Comment by jwintz** [comment]
You made your point. However, using `magit` together with `magit-flow`, I used to have a GitFlow segment that do not show up anymore. But it might be a minor mode, and as I use `diminish` for nearly all of them, I do not know whether it comes from `spaceline-all-the-icons` or from my setup.

![screen shot 2017-04-21 at 12 16 50](https://cloud.githubusercontent.com/assets/3277054/25273533/6b371be2-268c-11e7-8f5a-283e07eb75ed.png)

**Comment by domtronn** [comment]
Yep, you're right - I haven't implemented a `diminish` segment yet! It's another thing I don't personally use... But plan to add in in the future.

Again, any ideas/information about how you think it should be displayed would be great! 🙂 It'd also be great if you could show what your GitFlow segment used to look like and I can try and recreate it.

**Comment by jwintz** [comment]
You can see it here: https://github.com/d12frosted/homebrew-emacs-plus/pull/21.

**Comment by domtronn** [comment]
That does look like `diminish`, I'm going to raise a new issue for the `diminish` segment.
